ast/builtins.go

```diff
@@ -196,6 +196,56 @@ var FormatInt = &Builtin{
 	TargetPos: []int{2},
 }
 
+// IndexOf returns the index of a substring contained inside a string
+var IndexOf = &Builtin{
+	Name:      Var("indexof"),
+	NumArgs:   3,
+	TargetPos: []int{2},
+}
+
+// Substring returns the portion of a string for a given start index and a length.
+//   If the length is less than zero, then substring returns the remainder of the string.
+var Substring = &Builtin{
+	Name:      Var("substring"),
+	NumArgs:   4,
+	TargetPos: []int{3},
+}
+
+// Contains returns true if the search string is included in the base string
+var Contains = &Builtin{
+	Name:      Var("contains"),
+	NumArgs:   2,
+	TargetPos: []int{2},
```

Comment:
Contains should not have any outputs. TargetPos is probably not intended here--it can be removed.